pkg/securelaunch/helpers.go

```diff
@@ -0,0 +1,270 @@
+	// make sure dst is an absolute file path
+	if !filepath.IsAbs(dst) {
+		return "", fmt.Errorf("Error: dst =%s Not an absolute path ", dst)
+	}
+
+	// target is the full absolute path where []byte will be written to
+	target := dst
+
+	/* dst = /foo/bar/eventlog, check if /foo/bar/eventlog is a dir or file that exists..
+	     * if it exists as a file, this if loop is untouched.
+		 * if it exists as a dir, append defFileName to it
+	     * NOTE both dst = /foo/bar/ and dst=/foo/bar are considered dirs i.e. trailing "/" has no meaning..
+	*/
+	dstFound, is_dir, err := exists(dst)
+	if is_dir {
+		if !dstFound {
+			return "", fmt.Errorf("destination dir doesn't")
+		} else {
+			// User provided a dir, Use defFileName as target
+			Debug("No file name provided. Adding it now. old target=%s", target)
+			target = filepath.Join(dst, defFileName)
+			Debug("New target=%s", target)
+		}
+	}
+
+	Debug("target=%s", target)
+	err = ioutil.WriteFile(target, data, 0644)
+	if err != nil {
+		return "", fmt.Errorf("Could't write file to %s, err=%v", target, err)
+	}
+	Debug("WriteToFile exit w success data written to target=%s", target)
+	return target, nil
+}
+
+/*
+ * src and dst are already mounted file paths.
+ * src and dst are Absolute file paths
+ * defFileName is default dst file name, only used if user doesn't provide one.
+ * returns
+ * 1. target file path where file was written
+ * 2. error
+ */
+func WriteSrcFiletoDst(src, dst, defFileName string) (string, error) {
+	srcFileFound, _, err := exists(src)
+	if !srcFileFound {
+		return "", fmt.Errorf("File to be written doesnt exist:")
+	}
+
+	d, err := ioutil.ReadFile(src)
+	if err != nil {
+		return "", fmt.Errorf("Error reading src file %s, err=%v", src, err)
+	}
+
+	target, err := WriteToFile(d, dst, defFileName)
+	if err != nil {
+		return "", fmt.Errorf("WriteSrcFiletoDst failed err=%v", err)
+	}
+
+	Debug("WriteSrcFileToDst exit w success src=%s written to target=%s", src, target)
+	return target, nil
+}
+
+/*
+ * NOTE: Caller's responsbility to unmount this..use return var mountPath to unmount in caller.
+ *
+ * obtains absolute file path based on input entered by user in policy file.
+ * inputVal is of format <block device identifier>:<path>
+ * inputVal e.g sda:/boot/foo.go , 4qccd342-12zr-4e99-9ze7-1234cb1234c4:/bar/zyx.go
+ * rw_option = true --> mount is of type read-write, else read-only.
+ *
+ * This function
+ * parses user input.
+ * mounts device
+ * Get absolute file path on mounted device.
+ * does NOT Unmount device, use returned mountPath to unmount...
+ * returns filePath, mountPath, error in that order
+ */
+func GetMountedFilePath(inputVal string, rw_option bool) (string, string, error) {
+	s := strings.Split(inputVal, ":")
+	if len(s) != 2 {
+		return "", "", fmt.Errorf("%s: Usage: <block device identifier>:<path>", inputVal)
+	}
+
+	// s[0] can be sda or UUID. if UUID, then we need to find its name
+	var deviceId string = s[0]
+	if !strings.HasPrefix(deviceId, "sd") {
```

Comment:
is there a nicer way to recognize if this is a /dev device? what about /dev/hda, for example?

Reply:
Not sure.. any ideas ?

Reply:
I wonder if you could just do something like

```go
dev := filepath.Join("/dev", deviceID)
if _, err := os.Stat(dev); err == nil {
  // exists and is usable
} else {
  // does not exist or is not usable (permission error, e.g.)
}
```

Reply:
in linux everything is technically a file..
but then we have the other case of UUIDs..
I think the purpose of the code here is to just differentiate sda from 4PIOR-BLAH-BLAH-UUID
we don't care at this point if the device exists...
I would think that the downstream functions such as diskboot.FindDevice in this case should take care of the existence of the device or any errors...

Reply:
but the devtmpfs file system is very predictable. it never has a file like /dev/4PIOR-BLAH-BLAH-UUID (and if it does, it's likely the right thing to use anyway, right?)

FindDevice is going to be doing something like this anyway.

Otherwise, I'd say try one, then fall back to the other

```go
if err := FindDevice(...); err != nil {
  storage.PartitionsByFsUUID(..)
}
```

or try to check for something that would only be in a UUID.

Reply:
For the code above, we would need to strip out the /dev/ that FindDevice wants and PartitionsByFsUUID doesn't...

Reply:
I don't think so:

```go
d := filepath.Join("/dev", deviceID)
if err := FindDevice(d); os.IsNotExist(err) { 
  PartitionByFsUUID(deviceID)
```
?

Anyway, if you don't agree we can leave this for a future PR to improve.

Reply:
I don't get it sorry. FindDevice also mounts the device...and we want the device obtained from PartitionByFsUUID mounted too...